buffer.py

```diff
@@ -887,13 +907,21 @@ def paintEvent(self, event):
 
             painter.drawPixmap(rect, qpixmap)
 
+            # Draw an indicator for synctex position
+            if self.synctex_page_num == index + 1 and self.synctex_pos_y != None:
+                indicator_pos_y = self.synctex_pos_y * self.scale
+                self.draw_synctex_indicator(painter, 15, indicator_pos_y)
+
             render_y += render_height
 
         # Clean unused pixmap cache that avoid use too much memory.
         self.clean_unused_page_cache_pixmap()
-
         painter.restore()
 
+        # Clear synctex indicator after a timeout
+        if self.synctex_time != None and time.time() - self.synctex_time > 5:
+            self.reset_synctex_indicator()
+
```

Comment:
Don't test time duration in paintEvent, paintEvent only call when page need change.

QTimer is better solution.